controllers/feeds/getFeedsV2.js

```diff
@@ -94,6 +94,47 @@ const isValidActivity = async (item, conditions) => {
   return (item.final_score || 0) >= threshold;
 };
 
+const getActorFromLatestReaction = (latestReaction) => {
+  let postActors = new Set();
+  latestReaction?.upvotes?.forEach((upvote) => postActors.add(upvote.user_id));
+  latestReaction?.downvotes?.forEach((downvote) => postActors.add(downvote.user_id));
+  latestReaction?.comments?.forEach((comments) => postActors.add(comments.user_id));
+  latestReaction?.comment?.forEach((comment) => postActors.add(comment.user_id));
+  return Array.from(postActors);
+};
+
+const setKarmaScore = (latestReaction, karmaScores) => {
+  return latestReaction.map((reaction) => {
+    const user = karmaScores.find((user) => user.user_id === reaction.user_id);
+    if (reaction.latest_children) {
+      reaction.latest_children = addKarmaScoreToLatestReaction(
+        reaction.latest_children,
+        karmaScores
+      );
+    }
+    return {
+      ...reaction,
+      karma_score: roundingKarmaScore(user?.karma_score || 0)
+    };
+  });
+};
+
+const addKarmaScoreToLatestReaction = (latestReaction, karmaScores) => {
+  if (latestReaction?.upvotes) {
+    latestReaction.upvotes = setKarmaScore(latestReaction.upvotes, karmaScores);
+  }
+  if (latestReaction?.downvotes) {
+    latestReaction.downvotes = setKarmaScore(latestReaction.downvotes, karmaScores);
+  }
+  if (latestReaction?.comments) {
+    latestReaction.comments = setKarmaScore(latestReaction.comments, karmaScores);
+  }
+  if (latestReaction?.comment) {
+    latestReaction.comment = setKarmaScore(latestReaction.comment, karmaScores);
+  }
+  return latestReaction;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `addKarmaScoreToLatestReaction` is also directly modifying the `latestReaction` argument. Similar to the previous comment, consider creating a new object instead of mutating the existing one.



<!-- This is an auto-generated comment by OSS CodeRabbit -->